socs/agents/hwp_supervisor/agent.py

```diff
@@ -750,6 +795,89 @@ def sleep_until_complete(self, session=None, dt=1):
             time.sleep(dt)
 
 
+@contextmanager
+def ensure_grip_safety(hwp_state: HWPState, timeout: float = 60.):
```

Comment:
I would like to have some more logs in this function.
For example, 'hwp rotation frequency safety checks have passed', 'acu safety checks have passed', 'start blocking acu motion' etc.